### ISSUE TITLE
Enable TypeScript strict mode

### ISSUE THREAD

**Issue by pavish** [centerofci/mathesar#619]
## Description
[strictNullChecks](https://www.typescriptlang.org/tsconfig#strictNullChecks) are currently not enabled, as they tend to increase boilerplate condition checks. It may lead to the code getting more verbose and being forced to check `null` and `undefined`, when we know for certain those values will not occur in certain scenarios.

However, considering that our contributors would be working on small improvements, it is better to have it enabled to avoid unforeseen bugs. The trade-off seems worth it.

**Comment by seancolsen** [comment]
@pavish What do you think about enabling [strict](https://www.typescriptlang.org/tsconfig#strict) instead of strictNullChecks?

**Comment by pavish** [comment]
> What do you think about enabling strict instead of strictNullChecks?

@seancolsen Sounds good to me, we can enable strict. We can consider disabling specific rules if we run into issues in the future.

**Comment by seancolsen** [comment]
We made some progress towards this ticket in #826 but there's still quite a bit more work to do cleaning up errors before we can turn on strict mode.

It's a little hard do imagine finding the time to prioritize that clean-up work before our Alpha release.

Also, I'm noticing new code that will add more type errors once we turn on strict mode, so we might find ourselves chasing a moving target to some extent.

@pavish As a stopgap measure to address the "moving target" problem, I'm wondering how you would feel about the following approach, going forward:

1. Merge a PR which does the following, thereby closing this ticket:
    1. Turn on strict mode.
    1. Add `// @ts-ignore` comments to suppress all existing errors.
1. Open a new ticket, requesting the refactoring necessary to eliminate all `// @ts-ignore` statements.
1. Rely on strict mode being set to keep all our new code in conformance.

**Comment by pavish** [comment]
@seancolsen I think this approach sounds good. Let's move forward with it.
